core/src/main/java/org/sonarsource/sonarlint/core/issue/IssueServiceImpl.java

```diff
@@ -178,17 +176,36 @@ public CompletableFuture<Void> addComment(AddIssueCommentParams params) {
 
   @Override
   public CompletableFuture<ReopenIssueResponse> reopenIssue(ReopenIssueParams params) {
-    var issueId = params.getIssueId();
-    var issueUuidOptional = asUUID(issueId);
-    if (issueUuidOptional.isEmpty()) {
-      return CompletableFuture.completedFuture(new ReopenIssueResponse(false));
-    }
-    var issueUuid = issueUuidOptional.get();
     var configurationScopeId = params.getConfigurationScopeId();
-    var localOnlyIssueStore = localOnlyIssueStorageService.get();
-    return removeIssue(localOnlyIssueStore, configurationScopeId, issueUuid)
-      .thenApply(v -> localOnlyIssueStorageService.get().removeIssue(issueUuid))
-      .thenApply(ReopenIssueResponse::new);
+    var optionalBinding = configurationRepository.getEffectiveBinding(configurationScopeId);
+    return optionalBinding
+      .flatMap(effectiveBinding -> serverApiProvider.getServerApi(effectiveBinding.getConnectionId()))
+      .map(connection -> {
+        var binding = optionalBinding.get();
+        var projectServerIssueStore = storageService.binding(binding).findings();
+        var issueId = params.getIssueId();
+        boolean isServerIssue = projectServerIssueStore.containsIssue(issueId, false);
+        if (isServerIssue) {
+          return connection.issue().changeStatusAsync(issueId, Transition.REOPEN)
```

Comment:
Nitpick: The method starts to become big. I would suggest extracting this branch and the else to two new methods, something like `reopenLocalIssue` and `reopenServerIssue`

Reply:
Good point!